plugins/module_utils/vars.py

```diff
@@ -0,0 +1,261 @@
+            # Ignoring value of changes[k] when it None permit to avoid to
+            # update this item when it explicitly skipped with
+            # `merge_seq_by_index=True`.
+            if k not in keys_changes or modifs[k] is None:
+                new[k] = original[k]
+            elif self._both_are_assimilable_to_dict(original.get(k), modifs[k]):
+                new[k] = self._get_new_dict_with_modif_present(original[k], modifs[k])
+            elif self._both_are_assimilable_to_list(original.get(k), modifs[k]):
+                new[k] = self._get_new_list_with_modif_present(original[k], modifs[k])
+            else:
+                new[k] = modifs[k]
+        return new
+
+    def _get_new_dict_with_modif_absent(self, original, modifs):
+        # type: (Mapping, Mapping) -> dict
+        new = {}
+        for k, base_elem in original.items():
+            change_elem = modifs.get(k)
+            if self._both_are_assimilable_to_dict(base_elem, change_elem):
+                intersect_res = self._get_new_dict_with_modif_absent(base_elem, change_elem)
+                if len(intersect_res) > 0 or self._keep_empty:
+                    new[k] = intersect_res
+            elif self._both_are_assimilable_to_list(base_elem, change_elem):
+                intersect_res = self._get_new_list_with_modif_absent(base_elem, change_elem)
+                if len(intersect_res) > 0 or self._keep_empty:
+                    new[k] = intersect_res
+            elif not base_elem == change_elem and not (
+                    self._remove_none and change_elem is None and k in modifs.keys()):
+                new[k] = base_elem
+        return new
+
+    def _get_new_list_with_modif_present(self, original, modifs):
+        # type: (Sequence, Sequence) -> list
+        if self._merge_seq_by_index:
+            base_dict = self._convert_generic_sequence_to_dict(original)
+            changes_dict = self._convert_generic_sequence_to_dict(modifs)
+            res_dict = self._get_new_dict_with_modif_present(base_dict, changes_dict)
+            new = self._convert_generic_mapping_to_list(res_dict)
+        else:
+            new = original + [elem for elem in modifs if elem not in original]
+        return new
+
+    def _get_new_list_with_modif_absent(self, original, modifs):
+        # type: (Sequence, Sequence) -> list
+        if self._merge_seq_by_index:
+            base_dict = self._convert_generic_sequence_to_dict(original)
+            changes_dict = self._convert_generic_sequence_to_dict(modifs)
+            new = self._get_new_dict_with_modif_absent(base_dict, changes_dict)
+            new = self._convert_generic_mapping_to_list(new)
+        else:
+            new = [elem for elem in original if elem not in modifs]
+        return new
+
+    @staticmethod
+    def _both_are_assimilable_to_dict(original, modifs):
+        # type: (any, any) -> bool
+        return (isinstance(original, Mapping) and isinstance(modifs, Mapping))
```

Comment:
Extra brackes are not needed in Python (same below).
```suggestion
        return isinstance(original, Mapping) and isinstance(modifs, Mapping)
```